### ISSUE TITLE
Clear playlist does nothing

### ISSUE THREAD

**Issue by smarquespt** [bottiger/SoundWaves#49]
The "Clear Playlist" option in sliding menu does nothing. Playlist remains as is. Release 0.92 from f-droid

**Comment by bottiger** [comment]
Good point. "Clear playlist" is actually a bogus title since that's not what it does. I guess it should be caleld "Reset playlist" since it removes all "manually queued" items and reverts to showing the most recents episodes.

However, I am currently working on fixing the playlist (both bugs, workflow etc), so any kind of input would be appreciated.

**Comment by smarquespt** [comment]
Ideas:

1 - Some kind of sort options (by podcast, alfabetically...)
2 - Don´t automatically add episodes to the playlist. Leave them in subscritions and only add them to playlist after download. 
3 - Real clear playlist option (remove all episodes)
4- Remove (or not) bulk download from playlist and add it to podcast tab. This way we can bulk download for each podcast subscription. This way is a perfect integration with point 2 above.
5 - Ability to move episode to top/bottom and organize them the way we want. Move above/below as we want them to be in playlist.

**Comment by bottiger** [comment]
@smarquespt 

There are two primary aspect to the playlist refactoring I will do now.

1) Make the current filters more powerful
2) Make it easier.

My personal workflow works by glancing over the current list of recent podcast and just hit play on the one I want to hear - but I understand I'm the odd one here :) Many (all other?) people prefer a traditional playlist.

To address my point "Make the current filters more powerful".

* In the navigation drawer there is a playlist filter where you can choose which subscriptions the app will add to the playlist. I will add an extra option called "none". This will, for all intents and purposes, convert the playlist into a traditional playlist where you have to queue things manually. (addressing you point #1)
* The choice of filter (and everything else in the navigation drawer) will be persisted and remembered so you only set it when you want to change it.
* I will add some more sort options. (addressing you point #2)
* If you do not want the playlist to auto populate the "clear playlist" (which will be renamed to "reset playlist) will really clear the playlist. (addressing your point #3).
* I will also add an option to only show episodes which have been downloaded and can be played offline.

Addressing the point "Make it easier"

* The playlist will default to *not auto populate* itself with recent episodes
* If the playlist is empty (as it will be the first time a user opens the app) it will not just show a blank screen with an "empty playlist" label, but instead a toggle-button to turn on auto population and a short text explaining what it does and that it can be configured in the navigation drawer at any later time.
* Previously there was a small handle in the left side of the playlist where you could re-arrange it by drag-and-drop. However, many people hit it by mistake and didn't understand what was going on. I will make it possible to re-arrange the episodes in the playlist by long-clicking (and the drag&drop) them instead. (addressing you point #5).
* I will also make sure episodes are removed from the playlist if you listen to them and then start to listen to something else. Almost all apps I tried have the problem of old podcast you don't want to listen to filling up the playlist.

Regarding you point #4 I will also add bulk download to the subscription page, but right now I'm focusing on the playlist :)

Again - please let me know what you think.
